sky/packages/sky/lib/src/gestures/double_tap.dart

```diff
@@ -0,0 +1,164 @@
+        !_firstTap.isWithinTolerance(event, kDoubleTapTouchSlop))
+      return;
+    _stopDoubleTapTimer();
+    TapTracker tracker = new TapTracker(
+      event: event,
+      entry: GestureArena.instance.add(event.pointer, this)
+    );
+    _trackers[event.pointer] = tracker;
+    tracker.startTimer(() => _reject(tracker));
+    tracker.startTrackingPointer(router, handleEvent);
+  }
+
+  void handleEvent(PointerInputEvent event) {
+    TapTracker tracker = _trackers[event.pointer];
+    assert(tracker != null);
+    if (event.type == 'pointerup') {
+      if (_firstTap == null)
+        _registerFirstTap(tracker);
+      else
+        _registerSecondTap(tracker);
+    } else if (event.type == 'pointermove' &&
+        !tracker.isWithinTolerance(event, kTouchSlop)) {
+      _reject(tracker);
+    } else if (event.type == 'pointercancel') {
+      _reject(tracker);
+    }
+  }
+
+  void acceptGesture(int pointer) {}
+
+  void rejectGesture(int pointer) {
+    TapTracker tracker = _trackers[pointer];
+    // If tracker isn't in the list, check if this is the first tap tracker
+    if (tracker == null &&
+        _firstTap != null &&
+        _firstTap.pointer == pointer)
+      tracker = _firstTap;
+    // If tracker is still null, we rejected ourselves already
+    if (tracker != null)
+      _reject(tracker);
+  }
+
+  void _reject(TapTracker tracker) {
+    _trackers.remove(tracker.pointer);
+    tracker.entry.resolve(GestureDisposition.rejected);
+    _freezeTracker(tracker);
+    // If the first tap is in progress, and we've run out of taps to track,
+    // reset won't have any work to do.  But if we're in the second tap, we need
+    // to clear intermediate state.
+    if (_firstTap != null &&
+        (_trackers.isEmpty || tracker == _firstTap))
+      _reset();
+  }
+
+  void dispose() {
+    _reset();
+    router = null;
+  }
+
+  void _reset() {
+    _stopDoubleTapTimer();
+    if (_firstTap != null) {
+      // Note, order is important below in order for the resolve -> reject logic
+      // to work properly
+      TapTracker tracker = _firstTap;
+      _firstTap = null;
+      _reject(tracker);
+      GestureArena.instance.release(tracker.pointer);
+    }
+    _clearTrackers();
+  }
+
+  void _registerFirstTap(TapTracker tracker) {
+    _startDoubleTapTimer();
+    GestureArena.instance.hold(tracker.pointer);
+    // Note, order is important below in order for the clear -> reject logic to
+    // work properly.
+    _freezeTracker(tracker);
+    _trackers.remove(tracker.pointer);
+    _clearTrackers();
+    _firstTap = tracker;
+  }
+
+  void _registerSecondTap(TapTracker tracker) {
+    _firstTap.entry.resolve(GestureDisposition.accepted);
+    tracker.entry.resolve(GestureDisposition.accepted);
+    _freezeTracker(tracker);
+    _trackers.remove(tracker.pointer);
+    onDoubleTap?.call();
+    _reset();
+  }
+
+  void _clearTrackers() {
+    List<TapTracker> localTrackers = new List.from(_trackers.values);
```

Comment:
List<TapTracker>.from(...)

(if you run skyanalyzer it will catch this and the other occurrence of the same problem)